### PR TITLE
Reduce Visibility of methods

### PR DESCRIPTION
**Pull request by vanniktech**
Dunno if this is wanted or not.

### CODE REVIEW

File: moshi/src/main/java/com/squareup/moshi/LinkedHashTreeMap.java [hunk 1]
```diff
@@ -44,18 +44,18 @@ public int compare(Comparable a, Comparable b) {
...
-  Comparator<? super K> comparator;
-  Node<K, V>[] table;
-  final Node<K, V> header;
+  private Comparator<? super K> comparator;
+  private Node<K, V>[] table;
+  private final Node<K, V> header;
   int size = 0;
-  int modCount = 0;
-  int threshold;
+  private int modCount = 0;
```
Comment (NightlyNexus) [inline moshi/src/main/java/com/squareup/moshi/LinkedHashTreeMap.java]: `table` and `modCount` are accessed by the inner class `LinkedTreeMapIterator`, so synthetic accessor methods would be created for them if set to private.
Also, I think this class is copied from Gson, so it's probably best not to modify it here. (Just imo, I don't know.)

**Comment by swankjesse** [comment]
For package-private classes the `public` visibility is just a signal to the reader that this method is expected to be used by other classes in the package. We tend to avoid some uses of `private` visibility because it may make the compiler and runtime do more work (for private classes).

**Comment by swankjesse** [comment]
Thanks though.

File: moshi/src/main/java/com/squareup/moshi/LinkedHashTreeMap.java (hunk 1 above)
Comment (vanniktech) [inline moshi/src/main/java/com/squareup/moshi/LinkedHashTreeMap.java]: Uh right didn't see that. Everywhere else I thought of that.

**Pull request closed without merging**